src/fireedge/src/modules/features/General/slice.js

```diff
@@ -0,0 +1,324 @@
+import { cloneDeep, get, merge, pickBy, pullAt, set } from 'lodash'
+
+const initial = {
+  zone: 0,
+  appTitle: null,
+  isBeta: false,
+  withGroupSwitcher: false,
+  isLoading: false,
+  isFixMenu: false,
+  isUpdateDialog: false,
+  isFullMode: false,
+  tableViewMode: TABLE_VIEW_MODE.CARD,
+  breadcrumb: { parent: {}, child: {} },
+  loadedOsProfile: {},
+  upload: 0,
+  notifications: [],
+  selectedIds: [],
+  disabledSteps: [],
+  fieldPath: '',
+  modifiedFields: {},
+  defaultZone: 0,
+}
+
+const slice = createSlice({
+  name: 'general',
+  initialState: initial,
+  extraReducers: (builder) => {
+    builder
+      /* LOGOUT ACTION */
+      .addCase(logout, (state) => ({
+        ...initial,
+        // persistent app state
+        zone: state.defaultZone,
+        defaultZone: state.defaultZone,
+        appTitle: state.appTitle,
+        isBeta: state.isBeta,
+        withGroupSwitcher: state.withGroupSwitcher,
+      }))
+
+      /* UI ACTIONS */
+      .addCase(actions.fixMenu, (state, { payload }) => ({
+        ...state,
+        isFixMenu: !!payload,
+      }))
+      .addCase(actions.changeLoading, (state, { payload }) => ({
+        ...state,
+        isLoading: !!payload,
+      }))
+      .addCase(actions.changeAppTitle, (state, { payload: appTitle }) => {
+        const lowerAppTitle = String(appTitle).toLowerCase()
+        const isBeta = APPS_IN_BETA?.includes(lowerAppTitle)
+        const withGroupSwitcher = APPS_WITH_SWITCHER?.includes(lowerAppTitle)
+
+        return { ...state, appTitle, isBeta, withGroupSwitcher }
+      })
+      .addCase(actions.changeZone, (state, { payload }) => ({
+        ...state,
+        zone: payload,
+      }))
+      .addCase(actions.setSelectedIds, (state, { payload }) => {
+        state.selectedIds = payload
+      })
+      .addCase(actions.updateDisabledSteps, (state, { payload }) => {
+        state.disabledSteps = payload
+      })
+      .addCase(actions.setUpdateDialog, (state, { payload }) => {
+        state.isUpdateDialog = !!payload
+      })
+      .addCase(actions.setFullMode, (state, { payload }) => {
+        state.isFullMode = !!payload
+      })
+      .addCase(actions.setBreadcrumb, (state, { payload }) => {
+        state.breadcrumb = payload
+      })
+      .addCase(actions.setTableViewMode, (state, { payload }) => {
+        state.tableViewMode = payload
+      })
+      .addCase(actions.setLoadOsProfile, (state, { payload }) => {
+        const { stepId } = payload
+        const exists = get(state.loadedOsProfile, stepId, false)
+
+        if (!exists) {
+          set(state.loadedOsProfile, stepId, true)
+        }
+      })
+
+      .addCase(actions.resetLoadOsProfile, (state) => {
+        state.loadedOsProfile = {}
+      })
+
+      /* FIELD MODIFICATIONS */
+      .addCase(actions.setFieldPath, (state, { payload }) => {
+        state.fieldPath = payload
+      })
+      .addCase(actions.resetFieldPath, (state) => {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This route is missing authorization checks. Without proper authorization, sensitive routes may be exposed to unauthorized users, leading to potential data breaches or unauthorized actions. Ensure that authorization middleware or handlers are applied. For more information on missing authorization vulnerabilities, refer to [CWE-862](https://cwe.mitre.org/data/definitions/862.html).